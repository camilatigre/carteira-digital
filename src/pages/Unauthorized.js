import React from "react";
import Header from "../common/Header";
import { Link } from "react-router-dom";
import "../styles/pages/Unauthorized.css";
import Button from "@material-ui/core/Button";

const Unauthorized = () => {
  return (
    <div className="outsideBody">
      <Header whiteLogo={true} />
      <div id="unauthorized">
        <h1>401</h1>
        <p>
          Desculpe não foi possível encontrar seu usuário. Faça o Login
          novamente para continuar suas transações
        </p>

        <Link to="/">
          <Button variant="outlined" className="secondaryButton">
            Fazer login
          </Button>
        </Link>
      </div>
    </div>
  );
};

export default Unauthorized;
