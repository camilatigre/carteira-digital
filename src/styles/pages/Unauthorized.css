#unauthorized {
  color: #fff;
  text-align: center;
  display: block;
}

#unauthorized h1 {
  font-size: 15em;
  text-shadow: 4px 4px 4px #000;
}

#unauthorized p {
  font-size: 18px;
  margin: 20px 0 40px;
  font-weight: bold;
}

#unauthorized .secondaryButton:hover {
  background: #fff;
}
