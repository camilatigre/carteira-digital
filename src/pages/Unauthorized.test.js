import React from 'react';
import Unauthorized from './Unauthorized';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('<Unauthorized />', () => {
  it('Deveria renderizar a pagina de erro de autorização', () => {
    const component = shallow(<Unauthorized />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
