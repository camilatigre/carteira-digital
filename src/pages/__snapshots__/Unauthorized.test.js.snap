// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Unauthorized /> Deveria renderizar a pagina de erro de autorização 1`] = `
<div
  className="outsideBody"
>
  <Header
    whiteLogo={true}
  />
  <div
    id="unauthorized"
  >
    <h1>
      401
    </h1>
    <p>
      Desculpe não foi possível encontrar seu usuário. Faça o Login novamente para continuar suas transações
    </p>
    <Link
      to="/"
    >
      <WithStyles(ForwardRef(Button))
        className="secondaryButton"
        variant="outlined"
      >
        Fazer login
      </WithStyles(ForwardRef(Button))>
    </Link>
  </div>
</div>
`;
